Packages/com.unity.inputsystem/InputSystem/Plugins/PlayerInput/InputValue.cs

```diff
@@ -45,6 +82,36 @@ public TValue Get<TValue>()
         }
 
         ////TODO: proper message if value type isn't right
+        /// <summary>
+        /// Check if the action button is pressed.
+        /// </summary>
+        /// <returns>True if the button is activated over the button threshold. False otherwise</returns>
+        /// <remarks>
+        /// The following example shows how to read a value from a <see cref="PlayerInput"/> message.
+        ///
+        /// <example>
+        /// <code>
+        /// [RequireComponent(typeof(PlayerInput))]
+        /// public class MyPlayerLogic : MonoBehaviour
+        /// {
+        ///     private bool m_Fire;
+        ///
+        ///     // 'Fire' input action has been triggered.
+        ///     public void OnFire(InputValue value)
+        ///     {
+        ///         m_Fire = value.isPressed;
+        ///     }
+        ///
+        ///     public void OnUpdate()
```

Comment:
Should this be Update()? I think it is  a weird example to just store a bool to use the value in a separate function called every frame here. How about just calling another function from OnFire() (- eg Shoot()).

Reply:
My concern is: This could be confusing with the On... term, since it is used for Action messages of the PlayerInput as well.

Reply:
Hmm, I copied this from PlayerInput and its badly named there. I'll change this to call FireWeapon() and remove it here.